app/models/concerns/order_status.rb

```diff
@@ -93,6 +92,30 @@ module OrderStatus # rubocop:disable Metrics/ModuleLength
       event :close do
         transition received: :closed
       end
+
+      event :cancel do
+        old_status = ""
+
+        before do
+          old_status = status
+        end
+
+        transition all - [:canceled, :rejected] => :canceled
```

Comment:
Where is `all` coming from? Is that a method for enums, or the state machine gem or somewhere else? Since this is being included in `Order`, it looks at first glance to possibly be the same as `Order.all` which doesn't seem right...?

What if the order is in `"approved"` status? That doesn't seem to be handled by the case statement below.

Reply:
I made this decision based on https://github.com/amatsuda/stateful_enum#usage example:close.

This method does not apply to order. It applies to stateful_enum and is defined in https://github.com/amatsuda/stateful_enum/blob/master/lib/stateful_enum/machine.rb:103